### ISSUE TITLE
S3 Upload: The authorization mechanism you have provided is not supported.

### ISSUE THREAD

**Issue by gopeter** [buildkite/agent#179]
I'm trying to upload some artifacts to my own S3 bucket and get this message: 

```
2015-08-26 18:05:45 INFO   Found 28 files that match "coverage/**/*"
2015-08-26 18:05:45 DEBUG  Authorizing S3 credentials and finding bucket `MY_BUCKET` in region `eu-central-1`...
2015-08-26 18:05:46 FATAL  Failed to upload artifacts: Could not find bucket `MY_BUCKET` in region `eu-central-1` (The authorization mechanism you have provided is not supported. Please use AWS4-HMAC-SHA256.)
```

I'm using the latest buildkite-agent, installed with Homebrew.

**Comment by keithpitt** [comment]
That's weird! Maybe we need to update our AWS package. I'll have a test locally and see what I can find...

**Comment by keithpitt** [comment]
Is this bucket in Frankfurt by any chance?

**Comment by gopeter** [comment]
Yep, it's in Frankfurt :)

**Comment by keithpitt** [comment]
So it looks like buckets create in Frankfurt requires a newer version of authentication that the library we use doesn't support. We're switching out the AWS library we use to a different one which does have support. The next release of the agent should work with Frankfurt S3 buckets just fine - sorry about the hassle! And thanks for the report!

**Comment by keithpitt** [comment]
I'll push out a beta when it's done and if you'd be able to test @gopeter that'd be awesome :lollipop:

**Comment by gopeter** [comment]
Sorry for the late reply: thanks! :D

**Comment by jofan** [comment]
We are still getting this issue. Has the official AWS SDK still not been implemented?

**Comment by toolmantim** [comment]
@jofan sorry you're still having issues. Are you on Agent 3.0 (beta)? That uses the official AWS SDK.

**Comment by lox** [comment]
This is resolved in the v3 betas, but will still affect the 2.x series.

**Comment by jofan** [comment]
Thanks for your replies, we'll make sure to use v3 in our next attempt

**Comment by keithpitt** [comment]
This has been fixed in v3.0 of the agent since we're using the official AWS S3 library now!
